Fix end of game antag roles

flappybatpal opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- To automatically tag this PR, add the uppercase label(s) surrounded by brackets below, for example: [LABEL] -->
[BUG]

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Fixes critter antags showing as blank.
Gives poltergeist special role and adds to agimmicks.
Rewrites bad_traitorify and use it to handle antag critters.
Stops some antag roles showing up twice on the scoreboard?
Removes objectives from hostile critters as a bunch of objectives make little sense for them/could be impossible.

## Why's this needed? <!-- Describe why you think this should be added to the game. -->
Scoreboard should display roles.

flappybatpal commented:
It's very likely some of this is bad as it took a lot of trial and error, please advise.

ZeWaka commented on code/datums/abilities/wraith.dm:
```diff
@@ -728,6 +728,10 @@
 		lucky_dude.special_role = "poltergeist"
 		lucky_dude.dnr = 1
 		lucky_dude.transfer_to(P)
+		if (!lucky_dude.former_antagonist_roles.Find(lucky_dude.special_role))
```
use `in`, you don't need Find's idx

ZeWaka commented on code/modules/antagonists/changeling/abilities/hivemind.dm:
```diff
@@ -90,6 +90,8 @@
 				spider.mind.special_role = "handspider"
 			if (!(spider.mind in ticker.mode.Agimmicks))
 				ticker.mode.Agimmicks += spider.mind
+			if (!spider.mind.former_antagonist_roles.Find(spider.mind.special_role))
```
in

ZeWaka commented on code/modules/antagonists/changeling/abilities/hivemind.dm:
```diff
@@ -182,6 +184,8 @@
 				spider.mind.special_role = "eyespider"
 			if (!(spider.mind in ticker.mode.Agimmicks))
 				ticker.mode.Agimmicks += spider.mind
+			if (!spider.mind.former_antagonist_roles.Find(spider.mind.special_role))
```
in

ZeWaka commented on code/modules/antagonists/changeling/abilities/hivemind.dm:
```diff
@@ -275,6 +279,8 @@
 				spider.mind.special_role = "legworm"
 			if (!(spider.mind in ticker.mode.Agimmicks))
 				ticker.mode.Agimmicks += spider.mind
+			if (!spider.mind.former_antagonist_roles.Find(spider.mind.special_role))
```
in

ZeWaka commented on code/modules/antagonists/changeling/abilities/hivemind.dm:
```diff
@@ -359,6 +365,8 @@
 				crab.mind.special_role = "buttcrab"
 			if (!(crab.mind in ticker.mode.Agimmicks))
 				ticker.mode.Agimmicks += crab.mind
+			if (!crab.mind.former_antagonist_roles.Find(crab.mind.special_role))
```
in

flappybatpal commented:
Pruned the former_antagonist_roles additions. From looking at it, this did not seem suitable for changeling limbs and would be a new addition.
Changed the poltergeist to be added to agimmicks, looks like they weren't tracked as being an antag at all.

Changed the tips popup for generic antag.

ZeWaka commented on code/procs/jobprocs.dm:
```diff
@@ -730,22 +730,33 @@
+		if(!(H.mind in ticker.mode.traitors))
+			ticker.mode.traitors += H.mind
+	else
+		if (!(H.mind in ticker.mode.Agimmicks))
```
|=

ZeWaka commented on code/procs/jobprocs.dm:
```diff
@@ -730,22 +730,33 @@
+			new_objective.set_up()
+			H.mind.objectives += new_objective
+			H << browse(grabResource("html/traitorTips/traitorhardTips.html"),"window=antagTips;titlebar=1;size=600x400;can_minimize=0;can_resize=0")
+		if(!(H.mind in ticker.mode.traitors))
```
|=

ZeWaka commented on code/datums/abilities/wraith.dm:
```diff
@@ -728,6 +728,8 @@
 		lucky_dude.special_role = "poltergeist"
 		lucky_dude.dnr = 1
 		lucky_dude.transfer_to(P)
+		if (!(lucky_dude in ticker.mode.Agimmicks))
```
|=
